tests/python/gaia-ui-tests/gaiatest/apps/cost_control/regions/settings.py

```diff
@@ -62,6 +64,26 @@ def select_when_use_is_above_unit_and_value(self, unit, value):
         size.send_keys(value)
         self.marionette.find_element(*self._usage_done_button_locator).tap()
 
+    def tap_confirm_reset(self):
+        confirm_reset_button = Wait(self.marionette).until(
+            expected.element_present(*self._confirm_reset_button_locator))
+        Wait(self.marionette).until(expected.element_displayed(confirm_reset_button))
+
+        # Workaround for bug 1161088, where tapping on the button inside the app itself
+        # makes Marionette spew out NS_ERROR_NOT_INITIALIZED errors
+        x = confirm_reset_button.rect['x'] + confirm_reset_button.rect['width']//2
+        y = confirm_reset_button.rect['y'] + confirm_reset_button.rect['height']//2
+        self.marionette.switch_to_frame()
+        self.marionette.find_element(*self._screen_locator).tap(x=x, y=y)
+
+        # go back to iframe of usage app settings
+        Wait(self.marionette).until(lambda m: self.apps.displayed_app.src == self._src)
+        self.apps.switch_to_displayed_app()
+        frame = Wait(self.marionette).until(expected.element_present(
```

Comment:
We use the same logic in `__init__` to switch to the settings iFrame. I'd suggest to factorize it in a private function. It'll make `tap_confirm_reset` smaller.